activerecord/lib/active_record/connection_handling.rb

```diff
@@ -44,21 +44,14 @@ module ConnectionHandling
     #
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
-    def establish_connection(spec = nil)
+    def establish_connection(config = nil)
       raise "Anonymous class is not allowed." unless name
 
-      spec     ||= DEFAULT_ENV.call.to_sym
-      resolver =   ConnectionAdapters::ConnectionSpecification::Resolver.new configurations
-      # TODO: uses name on establish_connection, for backwards compatibility
-      spec     =   resolver.spec(spec, self == Base ? "primary" : name)
-      self.connection_specification_name = spec.name
+      config ||= DEFAULT_ENV.call.to_sym
+      spec_name = self == Base ? "primary" : name
+      self.connection_specification_name = spec_name
```

Comment:
Rereading this, it comes off as needlessly verbose. `specification` is pretty abstract and it feels like we're pushing implementation to the users' eyeballs. Users should only have to think in terms of `connections` — I think we should shorten this to `self.connection_name = 'primary'`.